com.hazelcast.client.cache.impl.nearcache.ClientCacheNearCacheCacheOnUpdateTest.with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss [API-2028] [API-2219]

_master_ (commit f7242148a89972cb339dbf927502859f3c2220dd)

Failed on redhat-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-RedHatJDK8/86/testReport/com.hazelcast.client.cache.impl.nearcache/ClientCacheNearCacheCacheOnUpdateTest/with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 100, NearCacheStatsImpl{ownedEntryCount=101, ownedEntryMemoryCost=11413, creationTime=1681440649914, hits=27073451, misses=49, ratio=55251940.8%, evictions=0, expirations=0, invalidations=9, invalidationRequests=13, lastPersistenceTime=0, persistenceCount=0, lastPersistenceDuration=0, lastPersistenceWrittenBytes=0, lastPersistenceKeyCount=0, lastPersistenceFailure=''}, NearCacheConfig{name=ClientCache, inMemoryFormat=BINARY, invalidateOnChange=true, timeToLiveSeconds=0, maxIdleSeconds=0, evictionConfig=EvictionConfig{size=2147483647, maxSizePolicy=ENTRY_COUNT, evictionPolicy=LRU, comparatorClassName=null, comparator=null}, cacheLocalEntries=false, localUpdatePolicy=CACHE_ON_UPDATE, preloaderConfig=NearCachePreloaderConfig{enabled=false, directory=, storeInitialDelaySeconds=600, storeIntervalSeconds=600}} expected:<0> but was:<49>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at com.hazelcast.client.cache.impl.nearcache.ClientCacheNearCacheCacheOnUpdateTest.with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss(ClientCacheNearCacheCacheOnUpdateTest.java:67)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)

```
</details>


<details><summary>Standard output:</summary>

```
Finished Running Test: with_invalidate_policy_concurrently_updated_near_cache_causes_misses in 11.105 seconds.
02:51:00,930  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [LifecycleService] Thread-24118 - hz.client_3479 [dev] [5.3.0-SNAPSHOT] HazelcastClient 5.3.0-SNAPSHOT (20230414 - f724214) is SHUTDOWN
02:51:00,930  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [LifecycleService] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
02:51:00,930 DEBUG |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [JobCoordinationService] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Added a shutting-down member: 8e16b837-5557-469f-882e-4f3296598f6a
02:51:00,930  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [MigrationManager] hz.ClientCacheNearCacheCacheOnUpdateTest_peaceful_blackburn.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5701 - 8e16b837-5557-469f-882e-4f3296598f6a this is handled
02:51:00,931  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [Node] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Shutting down connection manager...
02:51:00,931  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [Node] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Shutting down node engine...
02:51:00,931  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [CacheService] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Removed cache config: CacheConfig{name='ClientCache', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, dataPersistenceConfig=DataPersistenceConfig{enabled=false, fsync=false}, wanReplicationRef=null, merkleTreeConfig=MerkleTreeConfig{enabled=null, depth=10}}
02:51:00,934  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [NodeExtension] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Destroying node NodeExtension.
02:51:00,934  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [Node] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Hazelcast Shutdown is completed in 4 ms.
02:51:00,934  INFO |with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss| - [LifecycleService] Thread-24118 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss(com.hazelcast.client.cache.impl.nearcache.ClientCacheNearCacheCacheOnUpdateTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20230414', buildNumber=20230414, revision=f724214, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'with_cacheOnUpdate_policy_concurrently_updated_near_cache_does_not_cause_any_miss(com.hazelcast.client.cache.impl.nearcache.ClientCacheNearCacheCacheOnUpdateTest):'
02:50:45, accumulated pauses: 1075 ms, max pause: 477 ms, pauses over 1000 ms: 0
02:50:50, accumulated pauses: 374 ms, max pause: 186 ms, pauses over 1000 ms: 0
02:50:55, accumulated pauses: 328 ms, max pause: 102 ms, pauses over 1000 ms: 0
02:51:00, accumulated pauses: 434 ms, max pause: 428 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-RedHatJDK8/86/

--- a/hazelcast/src/test/java/com/hazelcast/client/cache/impl/nearcache/ClientCacheNearCacheCacheOnUpdateTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/cache/impl/nearcache/ClientCacheNearCacheCacheOnUpdateTest.java
@@ -43,6 +43,7 @@ import java.util.concurrent.atomic.AtomicBoolean;
 
 import static com.hazelcast.config.NearCacheConfig.LocalUpdatePolicy.CACHE_ON_UPDATE;
 import static com.hazelcast.config.NearCacheConfig.LocalUpdatePolicy.INVALIDATE;
+import static com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask.RECONCILIATION_INTERVAL_SECONDS;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
@@ -103,14 +104,14 @@ public class ClientCacheNearCacheCacheOnUpdateTest extends ClientNearCacheTestSu
         Runnable getter = () -> {
             int i = 0;
             while (!stop.get()) {
-                icacheOnClient.get(i++ % NUM_OF_KEYS);
+                icacheOnClient.get(++i % NUM_OF_KEYS);
             }
         };
 
         Runnable putter = () -> {
             int i = 0;
             while (!stop.get()) {
-                i = i++ % NUM_OF_KEYS;
+                i = ++i % NUM_OF_KEYS;
                 icacheOnClient.put(i, i);
             }
         };
@@ -140,7 +141,9 @@ public class ClientCacheNearCacheCacheOnUpdateTest extends ClientNearCacheTestSu
     private ICache<Integer, Integer> newNearCachedCache(NearCacheConfig.LocalUpdatePolicy localUpdatePolicy) {
         NearCacheConfig nearCacheConfig = getNearCacheConfig(localUpdatePolicy);
         ClientConfig clientConfig = getClientConfig()
-                .addNearCacheConfig(nearCacheConfig);
+                .addNearCacheConfig(nearCacheConfig)
+                // Disable the repairing task invalidations to prevent any interference with the test
+                .setProperty(RECONCILIATION_INTERVAL_SECONDS.getName(), "0");
 
         HazelcastClientProxy client = (HazelcastClientProxy) hazelcastFactory.newHazelcastClient(clientConfig);
         CachingProvider provider = new HazelcastClientCachingProvider(client);

